### ISSUE TITLE
blkid only detects some of my bcachefs disks (breaks bcachefs-tools in master)

### ISSUE THREAD

**Issue by YellowOnion** [util-linux/util-linux#3103]
```
/dev/nvme0n1p3: UUID="927137d3-8864-4756-9c7e-3b9d1efff07b" LABEL="root" BLOCK_SIZE="4096" UUID_SUB="3c28bb11-d68c-445f-ba7a-6657ad751d42" TYPE="bcachefs" PARTLABEL="primary" PARTUUID="2464e1fa-df0f-467c-99a7-b8b694786c02"
/dev/sdb: UUID="927137d3-8864-4756-9c7e-3b9d1efff07b" LABEL="root" BLOCK_SIZE="4096" UUID_SUB="5f4c7981-c0fa-4ad0-b847-9c8d4f1a64d1" TYPE="bcachefs"
/dev/sde: UUID="927137d3-8864-4756-9c7e-3b9d1efff07b" LABEL="root" BLOCK_SIZE="4096" UUID_SUB="bc8ebb9d-356e-4bba-a5ef-79d81a3a144a" TYPE="bcachefs"
```
This only lists 3 devices, while the file system has 5:

```
hdd (device 4):                  sda              rw
hdd (device 1):                  sdb              rw
hdd (device 3):                  sdc              rw
hdd (device 2):                  sde              rw
ssd (device 0):            nvme0n1p3              rw
```

I can upload my super-blocks if needed for debugging purposes.

**Comment by t-8ch** [comment]
Which version of util-linux are you using?
Does it also happen on latest master?
Can you provide debug logs for one of the failing one.

> I can upload my super-blocks if needed for debugging purposes.

That would be great.

**Comment by YellowOnion** [comment]
@t-8ch sorry I should have mentioned version:  v2.39.4

I can have a look at running blkid from master later, I really should be sleeping.

**Comment by YellowOnion** [comment]
@t-8ch is there a tool to extract super-blocks? or can I just dd the first 5MB?

**Comment by YellowOnion** [comment]
Here's the first 5MB off each disk.

[super-dump.tar.gz](https://github.com/user-attachments/files/16043300/super-dump.tar.gz)

**Comment by t-8ch** [comment]
@YellowOnion Thanks.

This is a duplicate of , fixed by 63ffa1c6e616fd391fddabc15314d3a735c62d50 on master and 2.40.1.

**Comment by YellowOnion** [comment]
@t-8ch thanks! Apparently version 2.40.1 or so was pulled from NixOS due to bugs, because of NixOS strict hash based dependancy system, it's non-trival for me to use 2.40.1 myself without rebuilding my entire stack...is there some chance for a backport or should I just hope that 2.40 bugs get fixed soon?

**Comment by t-8ch** [comment]
I'm not sure if v2.39 is still supported.

The revert to v2.39 in [0dbb16a0030a](https://github.com/NixOS/nixpkgs/commit/0dbb16a0030a6f30a94217c6ae3d48e5e95c717a) mentions "Too many issues with 2.40 for now" but the only issue that was report I found was #3011, which can be trivially worked around.
It would be nice to know what other issues are encountered.

**Comment by YellowOnion** [comment]
@t-8ch looks like most of the bugs have been reported: https://github.com/NixOS/nixpkgs/pull/309805 mostly darwin issues and static builds against musl, not much activity since may, so maybe those bugs have been fixed, 2.40 probably missed the merge window for NixOS 24.05, so not much activity while that was stabilized.
